std/experimental/allocator/gc_allocator.d

```diff
@@ -117,31 +117,31 @@ struct GCAllocator
     are `shared`.
     */
 
-    static shared GCAllocator instance;
+    static shared const GCAllocator instance;
```

Comment:
Why not immutable? Does it matter one way or the other?

Reply:
No, AFAICT. An immutable instance is automatically shared, but since GCAllocator contains no data I don't think it matters in this case.

Reply:
That's bizarre.

Reply:
@andralex @edi33416 ?

Reply:
The issue seems to be that when you have `static immutable GCAllocator instance` the following
`is(typeof(GCAllocator.instance) == shared)` is `false`, which is odd. 

If you have a look at [Segregator](https://github.com/dlang/phobos/blob/master/std/experimental/allocator/building_blocks/segregator.d#L260) it compares the small and large allocators against `shared` in order to decide how to `mixin` the `Impl` template struct.

With `immutable`, the is expression is `false` and thus the `shared` attribute is not applied.
I don't know why it's false; maybe a compiler bug? @andralex

Reply:
I had the same problems in the static assert. I agree, it looks like a compiler bug.

Reply:
@nordlow @n8sh so this isn't a compiler bug, it was just a misunderstanding.

The is expression `is(T == shared)` checks to see if `T` has `shared` among his type qualifiers, so it is ok to return `false` for `immutable`. `immutable` isn't `shared`, but it is convertible to it.

So the check should be `is(T : shared(T))`. There are a couple of allocators that are affected by this incomplete check; from the top of my head I know of `AffixAllocator` and `Segregator`, though there may be others.

I propose we add a `is(T == immutable)` to those, instead of using `:`, as this is, imho, simpler and clearer. @nordlow do you want to have a go at it?

Reply:
Sure. Ok to add it [this](https://github.com/dlang/phobos/pull/6440)?

Reply:
Probably a new PR would be better

Reply:
Which allocators should we update?

Reply:
Ok, I'll try that.

Reply:
@edi33416, can you have a go at this? My time is currently very spare.

Reply:
@nordlow I'll have at it after DConf

Reply:
Thanks.